zerver/lib/email_mirror.py

```diff
@@ -105,7 +105,8 @@ def create_missed_message_address(user_profile, message):
     if settings.EMAIL_GATEWAY_PATTERN == '':
         logging.warning("EMAIL_GATEWAY_PATTERN is an empty string, using "
                         "NOREPLY_EMAIL_ADDRESS in the 'from' field.")
-        return settings.NOREPLY_EMAIL_ADDRESS
+        from_email = "Zulip Missed Messages <%s>" % (settings.NOREPLY_EMAIL_ADDRESS,)
```

Comment:
We'll want to ensure this works even in the case where someone does put something like "amce noreply <noreply@acme.com>" as their settings.NOREPLY_EMAIL_ADDRESS.

Two options here are
* Have send_email take a from_email and from_name
* Add a format_email (or something, would have to think about the name) function to send_email, that takes a name and email.

In either case, we'd want to use email.utils.parseaddr to get the email part of noreply_email_address, regardless of what form they put the email in. 

from_email can always just default to "Zulip"